Source/WTF/wtf/text/StringCommon.h

```diff
@@ -522,7 +522,17 @@ template<typename SearchCharacterType, typename MatchCharacterType>
     requires(TriviallyComparableCodeUnits<SearchCharacterType, MatchCharacterType>)
 size_t findIgnoringASCIICase(std::span<const SearchCharacterType> source, std::span<const MatchCharacterType> matchCharacters, size_t startOffset = 0)
 {
-    ASSERT(source.size() >= matchCharacters.size());
+    if (matchCharacters.empty())
+        return 0;
```

Comment:
I don’t think this is the best value to return if matchCharacters is empty. I think we want to return `std::min(startOffset, source.size())` rather than `0`. Please add test cases for this, and if you think `0` is better, please explain why.
```suggestion
    if (matchCharacters.empty())
        return std::min(startOffset, source.size());
```

But I might suggest that if `startOffset` is `> source.size()` we should return `notFound`, even if we are looking for the empty string. Easiest way to do that is probably to remove the special check for empty string entirely (see my suggestion below). But another way would be to write something like this:
```suggestion
    if (matchCharacters.empty())
        return startOffset > source.size() ? notFound : startOffset;
```